addons/mail/models/mail_mail.py

```diff
@@ -292,12 +292,14 @@ def _send_prepare_values(self, partner=None):
         if partner:
             emails_normalized = tools.email_normalize_all(partner.email)
             if emails_normalized:
+                name = partner.name.replace('@', '') or "False"
                 email_to = [
-                    tools.formataddr((partner.name or "False", email or "False"))
+                    tools.formataddr((name, email or "False"))
```

Comment:
actually I would replace (all / most) manual formataddr by a custom ``contact_formataddr`` tool method that would do the @ strip and then call formataddr. Indeed we have several calls to formataddr in odoo to format recipients, I suppose we should be defensive and check all of them and protect them.